generator/src/main/java/com/faforever/neroxis/generator/MapGenerator.java

```diff
@@ -523,6 +525,23 @@ private void generate() {
         map.setName(mapName);
         map.setFolderName(mapName);
         map.setFilePrefix(mapName);
+
+        if (map.getTerrainShaderPath().equals(PBR_SHADER_NAME)) {
+            map.getBiome().terrainMaterials().getTexturePaths()[8] =
+                    Path.of("/maps", map.getFolderName(), "env", "texture", "heightRoughness.dds")
+                            .toString()
+                            .replace("\\", "/");
+            map.getBiome().terrainMaterials().getNormalPaths()[8] =
+                    map.getBiome().terrainMaterials().getCubeMaps().get(0).getPath();
+        }
+        map.getBiome().terrainMaterials().getTexturePaths()[9] =
+                Path.of("/maps", map.getFolderName(), "env", "texture", "mapwide.dds")
+                                       .toString()
+                                       .replace("\\", "/");
+        // This needs to be higher than the map size in ogrids to trigger all aspects of the terrain
+        // shader, so we set this to a sufficiently high value.
+        map.getBiome().terrainMaterials().getTextureScales()[9] = 10000;
```

Comment:
what do you mean?
81km maps are 4096x4096, so the value needs to be higher. 5000 would also suffice, but it doesn't hurt when it is 10,000.

Reply:
It work make more sense if it just needs to be bigger or the same size as the map to just set it to the mapsize or mapsize+1 rather than an arbitrary value